test/e2e/utils.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+func getContainerByName(containers []corev1.Container, name string) (corev1.Container, error) {
+	var container corev1.Container
+	for c := range containers {
+		if containers[c].Name == name {
+			container = containers[c]
+			return container, nil
+		}
+	}
+	return container, fmt.Errorf("container named %s not found", name)
+}
+
+// Gets K8S object from cluster, accepts wait options. Returns error if resource does not exist
+func getKubernetesResource(
+	ctx context.Context,
+	object k8s.Object,
```

Comment:
Pls use `NamespacedName` instead, 

https://github.com/kubernetes/kubernetes/blob/0d9dccfc6bc22f2370650cf5b8ad99cbba74ea23/staging/src/k8s.io/apimachinery/pkg/types/namespacedname.go#L27-L30

Alternative, use `ObjectKey`

https://github.com/kubernetes-sigs/controller-runtime/blob/6ad5c1dd4418489606d19dfb87bf38905b440561/pkg/client/interfaces.go#L32

Reply:
I meant to change the parameter from `k8s.Object` to `NamespacedName` or `ObjectKey`. You can even move this generic function into utils/utils.go.

Reply:
This PR is almost ready to be merged with this comment unresolved. Pls let me know if you want to resolve it in this PR or in a followup PR.

Reply:
Let me do it in a followup